### ISSUE TITLE
270/271 5010 support

### ISSUE THREAD

**Issue by Edwardiv1** [imsweb/x12-parser#68]
Is it possible to add support for the 270/271 5010 standard? So far I only see 4010 is supported.
Thanks

**Comment by angelaszek** [comment]
It's possible but we would need a configuration file similar to what is in the resources/mapping directory for the 4010 version.

**Comment by nonlinear5** [comment]
Hi there. I am also interested in the support for the 270/271 5010 standard, so I am trying to resurrect this discussion.

**Comment by chenw5000** [comment]
Hi, Dev

We are facing the same issue that the server returns 271 in 5010 version, but the x12-parser only support 4010.

Also, I looked into the mapping xml and found that in 271.4010.X092.A1.xml, the valid value for ISA11 is "U". I think for other files, it can be any character, and most of the time, we can use "^" to avoid collision.

<element xid="ISA11">
        <data_ele>I10</data_ele>
        <name>Interchange Control Standards Identifier</name>
        <usage>R</usage>
        <seq>11</seq>
        <valid_codes>
          <code>**U**</code>
        </valid_codes>
      </element>

Please help take a look.

Thanks

**Comment by westerly-lzh** [comment]
we also need 270/271 5010 support, we have the same issue, server returns 270/271 5010.

**Comment by Edwardiv1** [comment]
> It's possible but we would need a configuration file similar to what is in the resources/mapping directory for the 4010 version.

Hi @angelaszek , There's a fork that someone created to do this. Is there a problem with it preventing being merged?

Is there any guidance on how to create the configuration file?

Thanks

**Comment by angelaszek** [comment]
I don't see a pull request for that branch. 

I would use one of the existing configuration files as a template and use the specifications for that implementation to update it accordingly. Unfortunately I don't know of an easier way and it is a big undertaking.

**Comment by Edwardiv1** [comment]
@angelaszek You're right - looks like there's no pull request for it. I was referring to https://github.com/imsweb/x12-parser/issues/48 and the associated XML file at https://github.com/nddipiazza/x12-parser/blob/issue_48_add_271_5010_support/src/main/resources/mapping/271.5010.X279.A1.xml

Issue https://github.com/imsweb/x12-parser/issues/49 was blocking it, but it's from 2021, so I don't know how relevant it is any more.
Any idea whether the XML is correct or salvageable? Or is it better to start from scratch?

Thanks

**Comment by angelaszek** [comment]
I did a quick scan of that XML file and it looks correct in terms of the structure I would expect. I think it is a good place to start.

**Comment by anupchaudhari** [comment]
Hi there. I did try out the https://github.com/nddipiazza/x12-parser/tree/issue_48_add_271_5010_support fork. Unfortunately, it did not work for me. So I wrote the 270 & 271 5010 XML files from scratch with structure based on the 4010 file. I've been testing it with multiple cases from various payers and so far it seems to be working as expected. Have raised a PR - https://github.com/imsweb/x12-parser/pull/73

**Comment by atkawa7** [comment]
Hi @angelaszek,

I've been exploring and using this library, and so far it's working really well—I'm quite happy with it!

 I was wondering if it's possible to support additional transactions. To help with that, I built a scraper that uses data from [zenbridge.io/edi-dictionary](https://zenbridge.io/edi-dictionary)](https://zenbridge.io/edi-dictionary) to generate configuration files.

I'd be happy to contribute these configs to the repo if you're interested. Here's a sample for reference:
[5010_837.xml](https://gist.github.com/atkawa7/b958cb9192c08278a77bb556a0d57b91)](https://gist.github.com/atkawa7/b958cb9192c08278a77bb556a0d57b91)

Let me know what you think!

**Comment by ctmay4** [comment]
We have had multiple requests for 5010 support. If you would like to create a merge request we can take a look. Thanks!

**Comment by atkawa7** [comment]
I have created configs for all but the are missing (Interchange Control Headers and Group Functional Headers)[https://docs.oracle.com/cd/E19398-01/820-1275/agdaw/index.html] Maybe this could be a good starting point

[Archive.zip](https://github.com/user-attachments/files/20421423/Archive.zip)

**Comment by ctzen** [comment]
Hi, any news on getting 270/271 5010 support?

**Comment by atkawa7** [comment]
@ctzen Check the Archive zip and get the respective zip for now. They are many xml files in there
